demo/cpp/main.cpp

```diff
@@ -18,7 +18,7 @@
 //  Some required utilities
 class Logger : public nvinfer1::ILogger {
 public:
-    void log(Severity severity, const char* msg) override {
+    void log(Severity severity, const char* msg) noexcept override {
```

Comment:
Can we add `noexcept` according to the macro?
For example https://github.com/grimoire/amirstan_plugin/blob/07d32f82d0c0d6843233569ef2f1316ff55b170b/src/plugin/common/layer_base.h#L10

Than we can use it on both trt7 and trt8.